state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -321,13 +289,16 @@ impl L1Fetcher {
             async move {
                 while let Some(hash) = hash_rx.recv().await {
                     let tx = loop {
+                        let before = Instant::now();
```

Comment:
I'm wondering if we should move this above the loop. I think it'd be good to know the duration per tx instead of per fetch attempt. However. we would then also be inflating that metric with `FAILED_FETCH_RETRY_INTERVAL_S`, so I'm not convinced.

Reply:
yeah, that's the problem with hand-coded timings - one never knows what to measure... We should just learn to use a profiler, and measure everything... :-)